src/rgw/rgw_lc.cc

```diff
@@ -565,7 +587,9 @@ static int remove_expired_obj(
       << dendl;
     return ret;
   }
-  ret =  del_op->delete_obj(dpp, null_yield);
+
+  bool log_op = !remove_indeed || !zonegroup_lc_check(dpp, oc.driver->get_zone());
+  ret =  del_op->delete_obj(dpp, null_yield, log_op);
```

Comment:
am i correct that `!remove_indeed` means we'd create a delete marker instead of deleting an object version? is `remove_indeed` always true for non-versioned buckets? if so, this looks right

lifecycle on each zone would generate a delete marker with their own version id, but `log_op` means that data sync would replicate each of those to each zone. but what's the end result after replication? if there are two zones, do they end up with 2 delete markers or just 1? if 1, are they guaranteed to be the same one?

Reply:
> am i correct that `!remove_indeed` means we'd create a delete marker instead of deleting an object version? is `remove_indeed` always true for non-versioned buckets? if so, this looks right
> 
Yes, `!remove_indeed` means we'd create a delete marker instead of deleting an object version, and `remove_indeed` is always true for non-versioned buckets.

Reply:
sorry about "remove_indeed";  I've been carrying it along from early code.  At least we know what it means...

Reply:
https://tracker.ceph.com/issues/63799

Reply:
in this scenario, I'd have expected two delete marker versions on either zones, one from its own delete op, and the second syncing from the other zone. but I don't think we allow multiple delete markers for an object.

Reply:
the second delete marker creation will fail because we return an -ENOENT here in rgw.bucket_link_olh() if we already have a delete marker in 
https://github.com/ceph/ceph/blob/main/src/cls/rgw/cls_rgw.cc#L1695-L1702

Reply:
> the second delete marker creation will fail because we return an -ENOENT here in rgw.bucket_link_olh() if we already have a delete marker in https://github.com/ceph/ceph/blob/main/src/cls/rgw/cls_rgw.cc#L1695-L1702

Yes, this is exactly what I observed from my testing. The replication failed to create the delete-marker with the following error
`2023-12-13T01:57:17.297-0500 7f393fa49700  0 rgw async rados processor: ERROR: bucket shard callback failed. obj=file_4k[WFpr4YYECQ6z9qtSFkOdbS6v.1Cjg6.]. ret=(2) No such file or directory`

Reply:
@smanjara I have created [a dedicated issue tracker](https://tracker.ceph.com/issues/63799) for this delete-marker issue. I'll paste the above findings and discussion there.